lib/api/apiUtils/authorization/permissionChecks.js

```diff
@@ -372,6 +381,141 @@ function processBucketPolicy(requestType, bucket, canonicalID, arn, bucketOwner,
     return processedResult;
 }
 
+/**
+ *
+ * @param {BucketInfo} bucket - bucket object
+ * @param {Account} account - account object
+ * @param {array} apiNames - action names: operations to authorize
+ * @param {string} apiMethod - the main API call
+ * @param {number} inflight - inflight bytes
+ * @param {Logger} log - logger
+ * @param {function} callback - callback function
+ * @returns {boolean} - true if the quota is valid, false otherwise
+ */
+function validateQuotas(bucket, account, apiNames, apiMethod, inflight, log, callback) {
+    const bucketQuota = bucket.getQuota();
+    const accountQuota = account?.quota || 0;
+    let bucketQuotaExceeded = false;
+    let accountQuotaExceeded = false;
+
+    if ((bucketQuota <= 0 && accountQuota <= 0) || !ScubaClientInstance?.enabled) {
+        if (bucketQuota > 0 || accountQuota > 0) {
+            log.warn('quota is set for a bucket, but scuba is disabled', {
+                bucketName: bucket.getName(),
+            });
+        }
+        return callback(null);
+    }
+    const creationDate = new Date(bucket.getCreationDate()).getTime();
+
+    return async.forEach(apiNames, (apiName, done) => {
+        let shouldEvaluateCopyObject = false;
+        if (apiName === 'objectGet' && (apiMethod === 'objectCopy' || apiMethod === 'objectPutCopyPart')) {
```

Comment:
should we not have maps similar to `actionNeedQuotaCheck` for detecting this case?

Reply:
yes, but not in arsenal, in the cloudserver's consts I think, this is not cross service.

Reply:
theorically maybe, but practically it could make sense in arsenal as well:
* arsenal already includes lots of cloudserver (or backbeat, or vault) specific code ;
* it may make sense to keep this together with the other 2 variables, since this whole algorithm is based on these variables. The question is really "_would it make sense to modify just this new variable without considering if an update is required on the other 2_" ?

Reply:
After reviewing this, I'm not sure anymore it makes sense as a map: we need to pass two strings, the current API and the current action we evaluate, so better having a small function; a map won't make a lot of sense... The function will be added as a new function in arsenal.